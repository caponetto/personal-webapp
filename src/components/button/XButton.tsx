import XIcon from "@mui/icons-material/X";
import { useTranslation } from "react-i18next";
import { SocialButton } from "./SocialButton";

interface XButtonProps {
  disableSpacing?: boolean;
  url: string;
}

export function XButton(props: XButtonProps) {
  const { t } = useTranslation();
  return <SocialButton label={t("literal:x")} icon={<XIcon />} disableSpacing={props.disableSpacing} url={props.url} />;
}
