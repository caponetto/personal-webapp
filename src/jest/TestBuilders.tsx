import * as router from "react-router";
import { MediaItem } from "../schema";

export const DEFAULT_KEYWORD_SELECTION = {
  selectionMap: new Map(),
  onToggleSelection: jest.fn(),
  onClearSelection: jest.fn(),
  isAnySelected: false,
};

export const createMediaItem = (args: Partial<MediaItem>): MediaItem => ({
  title: args.title ?? "Title",
  keywordKeys: args.keywordKeys ?? [],
  kind: args.kind ?? "post",
  releaseDate: args.releaseDate ?? new Date(),
  publication: args.publication ?? "gitHub",
});

export const mockLocationTo = (args: Partial<router.Location>) => {
  jest.spyOn(router, "useLocation").mockImplementation(() => ({
    pathname: args.pathname ?? "/",
    search: args.search ?? "",
    hash: args.hash ?? "#",
    key: args.key ?? "default",
    state: args.state ?? undefined,
  }));
};
