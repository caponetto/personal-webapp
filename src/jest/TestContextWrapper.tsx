import { ReactElement } from "react";
import { I18nextProvider } from "react-i18next";
import { AppContextProvider } from "../context/AppContextProvider";
import i18n from "./I18nForTests";

export function usingTestingI18nContext(children: ReactElement) {
  return {
    wrapper: <I18nextProvider i18n={i18n}>{children}</I18nextProvider>,
  };
}

export function usingTestingAppContext(children: ReactElement) {
  return {
    wrapper: <AppContextProvider>{children}</AppContextProvider>,
  };
}

export function usingTestingContext(children: ReactElement) {
  return usingTestingI18nContext(usingTestingAppContext(children).wrapper).wrapper;
}
