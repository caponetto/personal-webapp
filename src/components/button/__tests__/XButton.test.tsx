import { render } from "@testing-library/react";
import { XButton } from "../XButton";
import { usingTestingI18nContext } from "../../../jest/TestContextWrapper";

describe("XButton", () => {
  it("should match the snapshot", () => {
    const { getByTestId } = render(usingTestingI18nContext(<XButton url="" />).wrapper);
    expect(getByTestId("x-button")).toMatchSnapshot();
  });
});
