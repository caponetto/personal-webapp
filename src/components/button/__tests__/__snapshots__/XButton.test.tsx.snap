// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`XButton should match the snapshot 1`] = `
<button
  aria-label="Open x"
  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1l55se7-MuiButtonBase-root-MuiIconButton-root"
  data-mui-internal-clone-element="true"
  data-testid="x-button"
  id="x-button"
  tabindex="0"
  type="button"
>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="XIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M18.244 2.25h3.308l-7.227 8.26 8.502 11.24H16.17l-5.214-6.817L4.99 21.75H1.68l7.73-8.835L1.254 2.25H8.08l4.713 6.231zm-1.161 17.52h1.833L7.084 4.126H5.117z"
    />
  </svg>
  <span
    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
  />
</button>
`;
